cl/donate/api_views.py

```diff
@@ -0,0 +1,215 @@
+            data = webhook_data["data"]
+            membership = {
+                "membershipId": data["id"],
+                "membershipName": data["membershipLevel"]["name"],
+            }
+
+        membership.update(
+            {
+                "accountId": data["accountId"],
+                "termEndDate": data["termEndDate"],
+            }
+        )
+
+        return membership
+
+    def _map_trigger_value(self, trigger_event: str) -> int:
+        """
+        Maps a string trigger event received from a Neon webhook to the
+        corresponding integer value representing the trigger event type in
+        the NeonWebhookEvents model.
+
+        Args:
+            trigger_event (str): The string representing the trigger event
+            type received in the Neon webhook payload.
+
+        Returns:
+            int: The integer value of the trigger event type according to
+            the NeonWebhookEvents model
+        """
+        match trigger_event:
+            case "createMembership":
+                trigger = NeonWebhookEvents.MEMBERSHIP_CREATION
+            case "updateMembership":
+                trigger = NeonWebhookEvents.MEMBERSHIP_UPDATE
+            case "editMembership":
+                trigger = NeonWebhookEvents.MEMBERSHIP_EDIT
+            case "deleteMembership":
+                trigger = NeonWebhookEvents.MEMBERSHIP_DELETE
+
+        return trigger
+
+    def _store_webhook_payload(self, webhook_data) -> None:
+        trigger = self._map_trigger_value(webhook_data["eventTrigger"])
+        if trigger != NeonWebhookEvents.MEMBERSHIP_DELETE:
+            membership_data = self._get_membership_data(webhook_data)
+        else:
+            membership_data = webhook_data["data"]["membership"]
+        NeonWebhookEvents.objects.create(
+            content=json.dumps(webhook_data, default=str),
+            account_id=membership_data.get("accountId", ""),
+            membership_id=membership_data["membershipId"],
+            trigger=trigger,
+        )
+
+    def _handle_membership_creation_or_update(self, webhook_data) -> None:
+        membership_data = self._get_membership_data(webhook_data)
+        user = self._get_member_record(membership_data["accountId"])
+        try:
+            neon_membership = user.membership
+            neon_membership.neon_id = membership_data["membershipId"]
+            neon_membership.level = NeonMembership.INVERTED[
+                membership_data["membershipName"]
+            ]
+            neon_membership.termination_date = membership_data["termEndDate"]
+            neon_membership.save()
+        except ObjectDoesNotExist:
+            NeonMembership.objects.create(
+                user=user,
+                neon_id=membership_data["membershipId"],
+                level=NeonMembership.INVERTED[
+                    membership_data["membershipName"]
+                ],
+                termination_date=membership_data["termEndDate"],
+            )
```

Comment:
We want as little in our `try` block as possible, so perhaps:

```suggestion
        try:
            neon_membership = user.membership
        except ObjectDoesNotExist:
            NeonMembership.objects.create(
                user=user,
                neon_id=membership_data["membershipId"],
                level=NeonMembership.INVERTED[
                    membership_data["membershipName"]
                ],
                termination_date=membership_data["termEndDate"],
            )
        else:
            neon_membership.neon_id = membership_data["membershipId"]
            neon_membership.level = NeonMembership.INVERTED[
                membership_data["membershipName"]
            ]
            neon_membership.termination_date = membership_data["termEndDate"]
            neon_membership.save()
```

Reply:
Thank you for your suggestion. I will implement this refactor.